applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -4,954 +4,868 @@
+            const IndexType number_of_conditions =
+                static_cast<IndexType>(r_conditions_array.size());
+            // generate map - use the "temp" array here
+            for (IndexType i = 0; i != number_of_local_dofs; i++)
+                temp[i] = mFirstMyId + i;
             Epetra_Map my_map(-1, number_of_local_dofs, temp, 0, mrComm);
-
-            //create and fill the graph of the matrix --> the temp array is reused here with a different meaning
-            Epetra_FECrsGraph Agraph(Copy, my_map, mguess_row_size);
-
-            Element::EquationIdVectorType EquationId;
-            ProcessInfo &CurrentProcessInfo = rModelPart.GetProcessInfo();
+            // create and fill the graph of the matrix --> the temp array is
+            // reused here with a different meaning
+            Epetra_FECrsGraph Agraph(Copy, my_map, mGuessRowSize);
+            Element::EquationIdVectorType equation_ids_vector;
+            ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
 
             // assemble all elements
-            for (typename ElementsArrayType::ptr_iterator it=rElements.ptr_begin(); it!=rElements.ptr_end(); ++it)
-            {
-                pScheme->EquationId(*it, EquationId, CurrentProcessInfo);
-
-                //filling the list of active global indices (non fixed)
-                unsigned int num_active_indices = 0;
-                for(unsigned int i=0; i<EquationId.size(); i++)
-                {
-                    temp[num_active_indices] =  EquationId[i];
+            for (IndexType i = 0; i < number_of_elements; ++i) {
+                auto it_elem = r_elements_array.begin() + i;
+                pScheme->EquationId(*(it_elem.base()), equation_ids_vector,
+                                    r_current_process_info);
+
+                // filling the list of active global indices (non fixed)
+                IndexType num_active_indices = 0;
+                for (IndexType i = 0; i < equation_ids_vector.size(); i++) {
+                    temp[num_active_indices] = equation_ids_vector[i];
                     num_active_indices += 1;
                 }
 
-                if(num_active_indices != 0)
-                {
-                    int ierr = Agraph.InsertGlobalIndices(num_active_indices,temp,num_active_indices, temp);
-                    KRATOS_ERROR_IF( ierr < 0 ) << "In " << __FILE__ << ":" << __LINE__ << ": Epetra failure in Graph.InsertGlobalIndices. Error code: " << ierr << std::endl;
+                if (num_active_indices != 0) {
+                    int ierr = Agraph.InsertGlobalIndices(
+                        num_active_indices, temp, num_active_indices, temp);
+                    KRATOS_ERROR_IF(ierr < 0)
+                        << "In " << __FILE__ << ":"
+                        << __LINE__ << ": Epetra failure in Graph.InsertGlobalIndices. Error code: "
+                        << ierr << std::endl;
```

Comment:
`__LINE` & `__FILE__` are included in [`KRATOS_ERROR`](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/code_location.h#L131)